include/flash.h

```diff
@@ -127,6 +137,28 @@ static inline int flash_write_protection_set(struct device *dev, bool enable)
 	return api->write_protection(dev, enable);
 }
 
+/**
+ *  @brief  Get layout of pages of flash.
```

Comment:
So you think about get-sector-boundaries-for-address. I Think this would be usefully for some cases. I'm not sure whether you mean "instead of API I proposed" or instead of the get-sector-boundaries-for-address implementation in the driver?